lib/matplotlib/cm.py

```diff
@@ -65,12 +65,39 @@ def _gen_cmap_d():
     # Generate reversed cmaps.
     for cmap in list(cmap_d.values()):
         rmap = cmap.reversed()
+        cmap._global = True
+        rmap._global = True
         cmap_d[rmap.name] = rmap
     return cmap_d
 
 
-cmap_d = _gen_cmap_d()
-locals().update(cmap_d)
+class _DeprecatedCmapDict(dict):
```

Comment:
Deriving from dict makes `_DeprecatedCmapDict` hold its own data.

` cmap_d = _DeprecatedCmapDict(_cmap_d)` below will make a copy, and both instances will run out of sync when the user starts to modify `cmap_d`.

I recommend deriving `_DeprecatedCmapDict` from `MutableMapping`, holding `_cmap_d` as a private attribute and forwarding all calls to `self._cmap_d`.